### PR TITLE
refactor: refactored code for correctness and staticcheck

### PR DESCRIPTION
**Pull request by docmerlin**
we refactored code for correctness and staticcheck.
We ignored ST1005 to keep backwards compatibility in log messages

This allows us to use staticcheck in this project, and cleans up the code.

### CODE REVIEW

File: services/load/service.go [hunk 1]
```diff
@@ -335,7 +335,7 @@ func (s *Service) loadTask(f string) error {
 	}
 	s.tasks[id] = true
 
-	return nil
+	return err
```
Comment (jeffreyssmith2nd) [inline services/load/service.go:338]: Won't err always be nil here?

File: services/load/service.go [hunk 2]
```diff
@@ -472,7 +474,7 @@ func (s *Service) loadVars(f string) error {
 	}
 	s.tasks[id] = true
 
-	return nil
```
Comment (jeffreyssmith2nd) [inline services/load/service.go:475]: Same thing here?

File: udf/server_test.go [hunk 3]
```diff
@@ -337,6 +337,7 @@ func TestUDF_SnapshotRestore(t *testing.T) {
 		req = <-u.Requests
 		if req == nil {
 			t.Error("expected snapshot message got nil")
+			return
```
Comment (jeffreyssmith2nd) [inline udf/server_test.go:340]: Alternatively, you could do t.Fatal(...)

File: udf/server_test.go [hunk 4]
```diff
@@ -352,6 +353,7 @@ func TestUDF_SnapshotRestore(t *testing.T) {
 		req = <-u.Requests
 		if req == nil {
 			t.Error("expected restore message got nil")
```
Comment (jeffreyssmith2nd) [inline udf/server_test.go:355]: Same here

File: tick/ast/lex.go [hunk 5]
```diff
@@ -513,7 +513,7 @@ func isSpace(r rune) bool {
 const durationUnits = "uµsmhdw"
 
 func isDurUnit(r rune) bool {
-	return strings.IndexRune(durationUnits, r) != -1
+	return !strings.ContainsRune(durationUnits, r)
```
Comment (jeffreyssmith2nd) [inline tick/ast/lex.go:516]: Should this be negated?

File: udf/server_test.go (hunk 3 above)
Comment (docmerlin) [reply udf/server_test.go:340]: Not supposed to do a t.Fatal in a goroutine that isn't the main one.

File: services/load/service.go (hunk 1 above)
Comment (docmerlin) [reply services/load/service.go]: The idea was to return err from the defer, but I removed that.

**Review by jeffreyssmith2nd — APPROVED** [review]

**Pull request merged**